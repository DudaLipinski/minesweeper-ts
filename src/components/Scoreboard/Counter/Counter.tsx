import { FC, memo } from 'react'
import styled from 'styled-components'

const Frame = styled.div`
  display: inline-block;
  padding: 0 0.3vw;
  color: #ec433c;
  border: 0.15vw inset;
  line-height: 2vw;
  letter-spacing: 0.08em;
  background: #333;
  text-shadow: 0 0 0.1vw #ec433c;
`

export interface CounterProps {
  children: string
}

export const Counter: FC<CounterProps> = memo(({ children }) => (
  <Frame>{children}</Frame>
))

Counter.displayName = 'Counter'
