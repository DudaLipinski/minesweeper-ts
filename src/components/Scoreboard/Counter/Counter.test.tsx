import { render } from '@testing-library/react'
import { Counter } from './Counter'

describe('Counter', () => {
  it('Should renders Counter correctly', () => {
    const { asFragment } = render(<Counter children="010" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
