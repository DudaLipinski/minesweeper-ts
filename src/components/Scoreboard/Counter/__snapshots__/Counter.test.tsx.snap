// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Should renders Counter 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe gmxfJB"
  >
    010
  </div>
</DocumentFragment>
`;

exports[`Counter Should renders Counter correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-bcXHqe gmxfJB"
  >
    010
  </div>
</DocumentFragment>
`;
